app/Mage.php

```diff
@@ -646,7 +646,8 @@ public static function exception($module = 'Mage_Core', $message = '', $code = 0
      */
     public static function throwException($message, $messageStorage = null)
     {
-        if ($messageStorage && ($storage = self::getSingleton($messageStorage))) {
+        if ($messageStorage) {
+            $storage = self::getSingleton($messageStorage);
```

Comment:
this is not an equivalent of the previous code.
the phpstan error sounds like a false positive. 

I believe that getSingleton can return false also (the phpdock on get singleton should be adapted.

Reply:
if we also have false positives I've no idea how to handle this stuff

Reply:
In this case its not phpstan fault as the phpdoc is wrong. 
but anyway Phpstan is just a tool, which also has bugs and sometimes reports false positives.
:)
Can you fix the getSingleton signature, so were good to go with this patch?

Reply:
true, I really missed that, it should be done